### Задача 1:
Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [9]:
import numpy as np
import matplotlib.pyplot as plot
from math import factorial

In [11]:
x = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [12]:
def mean_value(array): 
    return sum(array) / len(array)

In [16]:
print(f'Mean = {round(mean_value(x), 2)}')
a = round(mean_value(x), 2)

Mean = 65.3


In [17]:
np.mean(x)

65.3

In [20]:
def mean_sqr_dev(array):
    sqr_dev = (array - mean_value(array)) ** 2
    return (sum(sqr_dev) / (len(sqr_dev) - 1)) ** (1/2)

In [21]:
print(f'Mean squared deviation = {round(mean_sqr_dev(x), 2)}')

Mean squared deviation = 31.62


In [23]:
np.std(x, ddof = 1)

31.624607341019814

In [25]:
def dispersion(array, offset):
    sqr_dev = (array - mean_value(array)) ** 2
    return sum(sqr_dev) / (len(sqr_dev) - 1) if offset else sum(sqr_dev) / len(sqr_dev)

In [27]:
print(f'Biased variance = {round(dispersion(x, False),2)}\n'
    f'Unbiased variance = {round(dispersion(x, True),2)}')

Biased variance = 950.11
Unbiased variance = 1000.12


In [28]:
np.var(x,ddof=0)

950.11

In [29]:
np.var(x,ddof=1)

1000.1157894736842

### Solution: 
1. Mean = 65.3
2. Standard deviation = 31.62
3. Biased variance = 950.11
4. Unbiased variance = 1000.12

## Задача 2:
### В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Это событие модет произойти следующими способами:\
Р(А): m1 from the 1st box takes out 0 white balls and P(B) m2 from the 2nd box takes out 3 white balls;\
P(C): m3 from the 1st box takes out 2 white balls and P(F) m4 from the 2nd box takes out 2 white balls;\
P(E): m5 from the 1st box takes out 2 white balls and P(F) m6 from the 2nd box takes out 1 white ball

n1: amount taken out of the 1st box is 2 balls from 8: n1 Ckn = C28 \
n2: amount taken out of the 2nd box is 4 balls from 12: n2 Ckn = C412

The probability of 3 balls being white is determined by the sum of all occurances:  
P = P(A) * P(B) + P(C) * P(D) + P(E) * P(F)

#### Probability fo the first occurance taking place -
P(A) * P(B) = (C05*C23  / C28) * (C35*C17 / C412) 

In [31]:
def combinations(n,k):
    return factorial(n) / (factorial(k) * factorial(n-k))

In [35]:
P1 = (combinations(5,0) * combinations(3,2) / combinations(8,2) * combinations(5,3) * combinations(7,1) / combinations(12,4))

In [36]:
print(f'0 white balls from box 1 & 3 white balls from box 2 = {round(P1, 4)}')

0 white balls from box 1 & 3 white balls from box 2 = 0.0152


In [37]:
P2 = (combinations(5,1) * combinations(3,1) / combinations(8,2) * combinations(5,2) * combinations(7,2) / combinations(12,4))

In [38]:
print(f'1 white ball from box 1 & 2 white balls from box 2 = {round(P2, 4)}')

1 white ball from box 1 & 2 white balls from box 2 = 0.2273


In [39]:
P3 = (combinations(5,2) * combinations(3,0) / combinations(8,2) * combinations(5,1) * combinations(7,3) / combinations(12,4))

In [40]:
print(f'2 white balls from box 1 & 1 white ball from box 2 = {round(P3, 4)}')

2 white balls from box 1 & 1 white ball from box 2 = 0.1263


In [45]:
P = P1 + P2 + P3
print(f'The probability of 3 white balls is P = {round(P, 4)} = {round(P * 100, 2)}')

The probability of 3 white balls is P = 0.3687 = 36.87


### Answer: 

The probability of all 3 balls being white is 36.87%

## Задача 3. 
### На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?
5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?